### PR TITLE
Add Waveshare ESP32-S2-Pico

### PR DESCRIPTION
**Pull request by bill88t**
Disclaimer: I am not affiliated with waveshare in any way.

All board functions are working as intended.

My only concern is the USB_VID & USB_PID.
Waveshare hasn't provided official values and I stole the ones you see from Waveshare RP2040 Zero.
I have opened a support ticket to waveshare and this issue is also tracked in the [tinyuf2](https://github.com/adafruit/tinyuf2/pull/233) port pr.

image
(Ljinux is a wanna-be operating system that runs on top of circuitpython and is a very good benchmark)

image

### CODE REVIEW

**Comment by bill88t** [comment]
Board picture
PXL_20220727_131556599

**Comment by tannewt** [comment]
Thanks for the PR! You can probably get an Espressif PID here: https://github.com/espressif/usb-pids/blob/main/allocated-pids.txt

**Comment by bill88t** [comment]
From the support ticket, I got the official values.
Commiting them now

File: ports/espressif/boards/waveshare_esp32s2_pico/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,17 @@
+USB_VID = 0x239a
```
Comment (tannewt) [inline ports/espressif/boards/waveshare_esp32s2_pico/mpconfigboard.mk:1]: This is Adafruit's VID. This is in no way official from Waveshare. Can you link me to the support ticket?

Comment (bill88t) [reply ports/espressif/boards/waveshare_esp32s2_pico/mpconfigboard.mk:1]: As you wish, Attaching .eml file & screenshot.
image
https://github.com/adafruit/circuitpython/files/9244660/Waveshare.Re.ESP32-S2-Pico.USB_VID.USB_PID.-.Waveshare.support%40waveshare.com.-.2022-08-02.0420.zip
In zip cuz github doesn't allow .eml
Should you wish for me to reply to the support ticket, please let me know.

Comment (tannewt) [reply ports/espressif/boards/waveshare_esp32s2_pico/mpconfigboard.mk:1]: Screenshot is enough. Thanks. I won't merge this with an Adafruit VID. I'd suggest requesting one from Espressif.

Comment (ladyada) [reply ports/espressif/boards/waveshare_esp32s2_pico/mpconfigboard.mk:1]: @SSYYL @waveshare do not use adafruit VID for your products, they are allocated for adafruit products only

Comment (bill88t) [reply ports/espressif/boards/waveshare_esp32s2_pico/mpconfigboard.mk:1]: > I'd suggest requesting one from Espressif.

I have requested one in https://github.com/espressif/usb-pids/pull/55.

Comment (bill88t) [reply ports/espressif/boards/waveshare_esp32s2_pico/mpconfigboard.mk:1]: Got 0x810a (for cp) and 0x8109 (for uf2) from espressif.
chris

Comment (SSYYL) [reply ports/espressif/boards/waveshare_esp32s2_pico/mpconfigboard.mk]: Thank you @ladyada. I quite agree with you. I have told the colleague in charge of this board.

**Comment by bill88t** [comment]
I have no idea what is wrong with the github actions. Should changes be needed, let me know.

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**